src/shared/varlink-io.systemd.service.c

```diff
@@ -27,3 +27,29 @@ int varlink_method_ping(Varlink *link, JsonVariant *parameters, VarlinkMethodFla
 
         return varlink_reply(link, NULL);
 }
+
+int varlink_method_set_log_level(Varlink *link, JsonVariant *parameters, VarlinkMethodFlags flags, void *userdata) {
+        static const JsonDispatch dispatch_table[] = {
+                {"level", JSON_VARIANT_INTEGER, json_dispatch_int64, 0},
+                {}
+        };
+
+        int64_t level;
+        int r;
+
+        assert(link);
+        assert(parameters);
+
+        if (json_variant_elements(parameters) != 2)
+                return varlink_error_invalid_parameter(link, parameters);
+
+        r = json_dispatch(parameters, dispatch_table, NULL, 0, &level);
+        if (r < 0)
+                return r;
+
+        log_debug("Received io.systemd.system.SetLogLevel(%" PRIi64 ")", level);
+
+        log_set_max_level(LOG_PRI(level));
```

Comment:
could you also add something like this here:

```c
uid_t uid;
r = varlink_get_peer_uid(link, &uid);
if (r < 0)
        return r;
if (uid != getuid() && uid != 0)
        return varlink_error(link, VARLINK_ERROR_PERMISSION_DENIED);
```

i.e. no matter what, unpriv clients should not be able to change the debug level